### DISCUSSION TITLE
Changing Crypto Token Authentication Code in EJBCA CE 8

### DISCUSSION THREAD

**Discussion by merryo11** [Keyfactor/ejbca-ce#524] [General]
Hi All, I just wanted to know if after creating the crypto token with a password can be changed later on with a new password keeping in mind that the crypto token is in use against a CA and most importantly without disturbing any setting.

Thanks

**Comment by primetomas** [comment]
You can do that with the CLI, but not in the UI.
bin/ejbca.sh cryptotoken setpin --help

**Reply by merryo11** [reply]
Will that work in docker environment also because my front end is with docker on ubuntu 22 and mysql db on separate vlan.

**Reply by primetomas** [reply]
I _think_ you can sh into the container and use the CLI yes.
docker exec -ti container_id /bin/bash

**Reply by svenska-primekey** [reply]
That is what the docker exec command does. It takes you into the shell of the container to use the CLI.

**Comment by merryo11** [comment]
Thanks Tomas and Sven, I ran the commands with new test crypto and it did change the password/pin. I shall try it with my already running crypto that are hosting CA keys and update the results.
